### DISCUSSION TITLE
ERROR: read tcp 172.19.0.2:8554->172.19.0.3:39614: read: connection reset by peer

### DISCUSSION THREAD

**Discussion by vbbandeira** [bluenviron/mediamtx#463] [Questions]
Hello there folks,

I'm trying to stream an RTSP based on https://github.com/aler9/rtsp-simple-server. I was able to set up the stream, watch it running well in VLC, and so on.

Then, I've started to follow the tutorial on https://github.com/aws-samples/amazon-kinesis-video-streams-demos/tree/master/producer-cpp/docker-rtsp , and after running the `kvs_gstreamer_sample`, I got an error:

```
0:00:02.597636000  1035 0x55cfcc491320 INFO        GST_ERROR_SYSTEM gstelement.c:2145:gst_element_message_full_with_details:<source> posting message: Internal data stream error.
Error received from element source: Internal data stream error.
Debugging information: gstrtspsrc.c(5653): gst_rtspsrc_loop (): /GstPipeline:rtsp-kinesis-pipeline/GstRTSPSrc:source:
streaming stopped, reason not-negotiated (-4)
```

AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, STREAM_NAME and RTSP_URL are all set. The logs with `GST_DEBUG=4` can be found at https://gist.github.com/vbbandeira/3e7d407587bb74909328f86d52e89010 .

Can you guys please lemme know what is going on there?

Thanks!

**Comment by aler9** [comment]
moved here #464
